model/flow/address_test.go

```diff
@@ -163,11 +167,16 @@ func testAddressGeneration(t *testing.T) {
 			// this is only a sanity check of the implementation and not an exhaustive proof
 			r := uint64(rand.Intn(maxIndex - loop - 1))
 			state = chain.newAddressGeneratorAtIndex(r)
-			refAddress, err := state.NextAddress()
+			expectedIndex := r
+			refAddress, idx, err := state.NextAddress()
+			expectedIndex++
 			require.NoError(t, err)
+			assert.Equal(t, expectedIndex, idx)
 			for i := 0; i < loop; i++ {
-				address, err := state.NextAddress()
+				address, idx, err := state.NextAddress()
+				expectedIndex++
 				require.NoError(t, err)
+				assert.Equal(t, expectedIndex, idx)
```

Comment:
`r+1+i` seems easier to understand (at least to me 😅 )
```suggestion
				assert.Equal(t, r+1+i, idx)
```